python/hail/table.py

```diff
@@ -403,8 +415,9 @@ def parallelize(cls, rows, schema=None, key=None, n_partitions=None):
                 Env.hc()._jhc, rows.dtype._to_json(rows.value),
                 rows.dtype.element_type._jtype, joption(key), joption(n_partitions)))
 
-    @typecheck_method(keys=nullable(oneof(str, Expression)))
-    def key_by(self, *keys) -> 'Table':
+    @typecheck_method(keys=nullable(oneof(str, Expression)),
+                      named_keys=Expression)
+    def key_by(self, *keys, **named_keys) -> 'Table':
```

Comment:
I think Table.key_by should be the place where we have the most detailed documentation about how the key_by family of methods works. Some of the docs from k_r_b / k_c_b should be moved here